packages/components/src/DateTimePickers/hooks/useInputPickerHandlers.js

```diff
@@ -0,0 +1,82 @@
+import { useState } from 'react';
+import keycode from 'keycode';
+
+
+export default function useInputPikcerHandlers(props) {
+	const [showPicker, setPickerVisibility] = useState(false);
+	const [picked, setPicked] = useState(false);
+
+	function openPicker() {
+		setPickerVisibility(true);
+	}
+	function closePicker() {
+		setPickerVisibility(false);
+	}
+	function onBlur(event) {
+		setPicked(false);
+		closePicker();
+		if (props.onBlur) {
+			props.onBlur(event);
+		}
+	}
+	function onChange(event, payload, inputRef) {
+		props.onChange(event, payload);
+		if (
+			(payload.origin !== 'INPUT')
+		) {
+			inputRef.focus();
+			setPicked(true);
+			closePicker();
+		}
+	}
+
+	function onClick() {
+		openPicker();
+	}
+
+	function onFocus() {
+		if (!picked) {
+			openPicker();
+		}
+	}
+	function onKeyDown(event, inputRef, containerRef) {
+		switch (event.keyCode) {
+			case keycode.codes.esc:
+				inputRef.focus();
+				closePicker();
+				break;
+			case keycode.codes.down:
+				if (event.target !== inputRef) {
+					return;
+				}
+
+				if (this.state.showPicker) {
+					// TODO: focuseOnTime();
+				} else {
+					openPicker();
+				}
+				break;
+			default:
+				break;
+		}
+	}
+	function getPopperPlacement(input) {
```

Comment:
Not sure, but I think popper.js manages that already.
Can you remove that and check if the placement is ok ?

Reply:
Still need this for InputDatePicker.
if removed, it will fall to default placement 'bottom', which align the picker to middle of input.
and we need 'bottom-start' or 'bottom-end' which align picker to left/right of input.
![placement](https://user-images.githubusercontent.com/3214228/63818972-0b096600-c976-11e9-9378-6dc587c6e3b7.png)

Reply:
In fact what I meant is to set the placement to what we want as default (here bottom-start) and see if popper manages the placement depending on the space

Reply:
I just tried, and popper doesn't auto adjust placement base on space. If I set placement="bottom-start", it always align start of input. even for the Input on the right top corner in picture above.